packages/jellyfish-testing/src/icxorderbook.ts

```diff
@@ -0,0 +1,242 @@
+import { Testing } from './index'
+import BigNumber from 'bignumber.js'
+import { icxorderbook } from '@defichain/jellyfish-api-core'
+
+export class TestingICX {
+  public symbolDFI: string
+  public symbolBTC: string
+  public accountDFI: string
+  public accountBTC: string
+  public poolOwner: string
+  public idDFI: string
+  public idBTC: string
+  public ICX_TAKERFEE_PER_BTC: BigNumber
+  public DEX_DFI_PER_BTC_RATE: BigNumber
+
+  constructor (
+    private readonly testing: Testing
+  ) {
+    this.accountDFI = ''
+    this.accountBTC = ''
+    this.poolOwner = ''
+    this.idDFI = ''
+    this.idBTC = ''
+    this.ICX_TAKERFEE_PER_BTC = new BigNumber(0)
+    this.DEX_DFI_PER_BTC_RATE = new BigNumber(0)
+    this.symbolDFI = 'DFI'
+    this.symbolBTC = 'BTC'
+  }
+
+  async setup (): Promise<void> {
+    await this.createAccounts()
+    await this.createBTCToken()
+    await this.initializeTokensIds()
+    await this.mintBTCtoken(100)
+    await this.fundAccount(this.accountDFI, this.symbolDFI, 500)
+    await this.fundAccount(this.accountBTC, this.symbolDFI, 10) // for fee
+    await this.createBTCDFIPool()
+    await this.addLiquidityToBTCDFIPool(1, 100)
+    await this.setTakerFee(new BigNumber(0.001))
+  }
+
+  async createAccounts (): Promise<void> {
+    this.accountDFI = await this.testing.container.getNewAddress()
+    this.accountBTC = await this.testing.container.getNewAddress()
+  }
+
+  async createBTCToken (): Promise<void> {
+    await this.testing.token.create({
+      symbol: this.symbolBTC,
+      name: this.symbolBTC,
+      isDAT: true,
+      mintable: true,
+      tradeable: true,
+      collateralAddress: this.accountBTC
+    })
+    await this.testing.container.generate(1)
+  }
+
+  async initializeTokensIds (): Promise<void> {
+    let tokenInfo = await this.testing.container.call('gettoken', [this.symbolBTC])
```

Comment:
can  use `this.testing.rpc.token.getToken()` here?
It's the same thing underneath, but good as a standard. so for RPCs that are not available yet on the jellyfish side, we could use `this.testing.container.call()`, otherwise `this.testing.rpc.`

Reply:
https://github.com/DeFiCh/jellyfish/issues/431